include/nbl/asset/filters/CBlitUtilities.h

```diff
@@ -77,120 +97,103 @@ class CBlitUtilities : public IBlitUtilities
-		computeForAxis(asset::IImage::ET_2D, scaledKernelY);
-		computeForAxis(asset::IImage::ET_3D, scaledKernelZ);
+		computeForAxis(asset::IImage::ET_1D, std::get<0>(kernels), windowSize.x);
+		computeForAxis(asset::IImage::ET_2D, std::get<1>(kernels), windowSize.y);
+		computeForAxis(asset::IImage::ET_3D, std::get<2>(kernels), windowSize.z);
 
 		return true;
 	}
 
-	static inline core::vectorSIMDi32 getRealWindowSize(const IImage::E_TYPE inImageType,
-		const CScaledImageFilterKernel<KernelX>& kernelX,
-		const CScaledImageFilterKernel<KernelY>& kernelY,
-		const CScaledImageFilterKernel<KernelZ>& kernelZ)
+	static inline convolution_kernels_t getConvolutionKernels(
+		const core::vectorSIMDu32&	inExtent,
+		const core::vectorSIMDu32&	outExtent,
+		ReconstructionFunctionX&&	reconstructionX		= ReconstructionFunctionX(),
+		ResamplingFunctionX&&		resamplingX			= ResamplingFunctionX(),
+		ReconstructionFunctionY&&	reconstructionY		= ReconstructionFunctionY(),
+		ResamplingFunctionY&&		resamplingY			= ResamplingFunctionY(),
+		ReconstructionFunctionZ&&	reconstructionZ		= ReconstructionFunctionZ(),
+		ResamplingFunctionZ&&		resamplingZ			= ResamplingFunctionZ())
 	{
-		core::vectorSIMDi32 last(kernelX.getWindowSize().x, 0, 0, 0);
+		// Stretch and scale the resampling kernels.
+		// we'll need to stretch the kernel support to be relative to the output image but in the input image coordinate system
+		// (if support is 3 pixels, it needs to be 3 output texels, but measured in input texels)
+
+		const auto rcp_c2 = core::vectorSIMDf(inExtent).preciseDivision(core::vectorSIMDf(outExtent));
+
+		resamplingX.stretchAndScale(rcp_c2.x);
+		resamplingY.stretchAndScale(rcp_c2.y);
+		resamplingZ.stretchAndScale(rcp_c2.z);
+
+		return {
+			convolution_kernel_x_t({std::move(reconstructionX), std::move(resamplingX)}),
+			convolution_kernel_y_t({std::move(reconstructionY), std::move(resamplingY)}),
+			convolution_kernel_z_t({std::move(reconstructionZ), std::move(resamplingZ)}) };
+	}
+
+	static inline core::vectorSIMDi32 getWindowSize(
+		const asset::IImage::E_TYPE	inImageType,
+		const convolution_kernels_t& kernels)
+	{
+		core::vectorSIMDi32 windowSize(std::get<0>(kernels).getWindowSize(), 0, 0, 0);
 		if (inImageType >= IImage::ET_2D)
-			last.y = kernelY.getWindowSize().y;
-		if (inImageType >= IImage::ET_3D)
-			last.z = kernelZ.getWindowSize().z;
-		return last;
+			windowSize.y = std::get<1>(kernels).getWindowSize();
+		if (inImageType == IImage::ET_3D)
+			windowSize.z = std::get<2>(kernels).getWindowSize();
+
+		return windowSize;
 	}
 
-	template <typename lut_value_type = typename KernelX::value_type>
-	static inline core::vectorSIMDu32 getScaledKernelPhasedLUTAxisOffsets(const core::vectorSIMDu32& phaseCount, const core::vectorSIMDi32& real_window_size)
+	template <typename LutDataType>
+	static inline core::vectorSIMDu32 getScaledKernelPhasedLUTAxisOffsets(const core::vectorSIMDu32& phaseCount, const core::vectorSIMDi32& windowSize)
```

Comment:
this is fine to keep as is, because weights eventually get compressed

So this has no bearing
https://github.com/Devsh-Graphics-Programming/Nabla/pull/435/files#r1196525176

(you can compress "in place" like a memmove)